tests/framework/utils_cpu_templates.py

```diff
@@ -52,6 +52,12 @@ def get_supported_custom_cpu_templates():
             return INTEL_TEMPLATES
         case CpuVendor.INTEL, CpuModel.INTEL_ICELAKE:
             return set(INTEL_TEMPLATES) - {"T2S"}
+        case CpuVendor.INTEL, CpuModel.INTEL_SAPPHIRE_RAPIDS:
+            # Intel AMX is only supported on kernel 5.17+. KVM does not support
+            # related CPUID range.
+            if host_linux >= (5, 17):
+                return ["SPR_TO_T2_6.1"]
+            return ["SPR_TO_T2_5.10"]
```

Comment:
mh, but T2 doesn't support AMX at all, so shouldn't it not matter either way? It would have to be disabled on >=5.17 anyway. Or would a template that unconditionally disables AMX fail to apply on 5.10 because KVM doesnt understand the leaves/bits?

Reply:
> Or would a template that unconditionally disables AMX fail to apply on 5.10 because KVM doesnt understand the leaves/bits?

yes. KVM doesn't understand and returns an error if attempt to modify those CPUID leaves.

Reply:
sadness